// bpmn-js/lib/features/modeling/behavior/index.js
import AdaptiveLabelPositioningBehavior from 'bpmn-js/lib/features/modeling/behavior/AdaptiveLabelPositioningBehavior';
import AppendBehavior from 'bpmn-js/lib/features/modeling/behavior/AppendBehavior';
import AssociationBehavior from 'bpmn-js/lib/features/modeling/behavior/AssociationBehavior';
import AttachEventBehavior from 'bpmn-js/lib/features/modeling/behavior/AttachEventBehavior';
import BoundaryEventBehavior from 'bpmn-js/lib/features/modeling/behavior/BoundaryEventBehavior';
import CompensateBoundaryEventBehavior from 'bpmn-js/lib/features/modeling/behavior/CompensateBoundaryEventBehavior';
import CreateBehavior from 'bpmn-js/lib/features/modeling/behavior/CreateBehavior';
import CreateDataObjectBehavior from 'bpmn-js/lib/features/modeling/behavior/CreateDataObjectBehavior';
import CreateParticipantBehavior from 'bpmn-js/lib/features/modeling/behavior/CreateParticipantBehavior';
import DataInputAssociationBehavior from 'bpmn-js/lib/features/modeling/behavior/DataInputAssociationBehavior';
import DataStoreBehavior from 'bpmn-js/lib/features/modeling/behavior/DataStoreBehavior';
import DeleteLaneBehavior from 'bpmn-js/lib/features/modeling/behavior/DeleteLaneBehavior';
import DetachEventBehavior from 'bpmn-js/lib/features/modeling/behavior/DetachEventBehavior';
import DropOnFlowBehavior from 'bpmn-js/lib/features/modeling/behavior/DropOnFlowBehavior';
import EventBasedGatewayBehavior from 'bpmn-js/lib/features/modeling/behavior/EventBasedGatewayBehavior';
import FixHoverBehavior from 'bpmn-js/lib/features/modeling/behavior/FixHoverBehavior';
import GroupBehavior from 'bpmn-js/lib/features/modeling/behavior/GroupBehavior';
import ImportDockingFix from 'bpmn-js/lib/features/modeling/behavior/ImportDockingFix';
import IsHorizontalFix from 'bpmn-js/lib/features/modeling/behavior/IsHorizontalFix';
import LabelBehavior from 'bpmn-js/lib/features/modeling/behavior/LabelBehavior';
import LayoutConnectionBehavior from 'bpmn-js/lib/features/modeling/behavior/LayoutConnectionBehavior';
import MessageFlowBehavior from 'bpmn-js/lib/features/modeling/behavior/MessageFlowBehavior';
import NonInterruptingBehavior from 'bpmn-js/lib/features/modeling/behavior/NonInterruptingBehavior';
import RemoveEmbeddedLabelBoundsBehavior from 'bpmn-js/lib/features/modeling/behavior/RemoveEmbeddedLabelBoundsBehavior';
import RemoveElementBehavior from 'bpmn-js/lib/features/modeling/behavior/RemoveElementBehavior';
import RemoveParticipantBehavior from 'bpmn-js/lib/features/modeling/behavior/RemoveParticipantBehavior';
import ReplaceConnectionBehavior from 'bpmn-js/lib/features/modeling/behavior/ReplaceConnectionBehavior';
import ReplaceElementBehaviour from 'bpmn-js/lib/features/modeling/behavior/ReplaceElementBehaviour';
import ResizeBehavior from 'bpmn-js/lib/features/modeling/behavior/ResizeBehavior';
import ResizeLaneBehavior from 'bpmn-js/lib/features/modeling/behavior/ResizeLaneBehavior';
import RootElementReferenceBehavior from 'bpmn-js/lib/features/modeling/behavior/RootElementReferenceBehavior';
import SpaceToolBehavior from 'bpmn-js/lib/features/modeling/behavior/SpaceToolBehavior';
import SubProcessPlaneBehavior from 'bpmn-js/lib/features/modeling/behavior/SubProcessPlaneBehavior';
import SubProcessStartEventBehavior from 'bpmn-js/lib/features/modeling/behavior/SubProcessStartEventBehavior';
import TextAnnotationBehavior from 'bpmn-js/lib/features/modeling/behavior/TextAnnotationBehavior';
import ToggleCollapseConnectionBehaviour from 'bpmn-js/lib/features/modeling/behavior/ToggleCollapseConnectionBehaviour';
import ToggleElementCollapseBehaviour from 'bpmn-js/lib/features/modeling/behavior/ToggleElementCollapseBehaviour';
import UnclaimIdBehavior from 'bpmn-js/lib/features/modeling/behavior/UnclaimIdBehavior';
import UnsetDefaultFlowBehavior from 'bpmn-js/lib/features/modeling/behavior/UnsetDefaultFlowBehavior';
import UpdateFlowNodeRefsBehavior from 'bpmn-js/lib/features/modeling/behavior/UpdateFlowNodeRefsBehavior';

/**
 * @type { import('didi').ModuleDeclaration }
 */
export default {
  __init__: [
    'adaptiveLabelPositioningBehavior',
    'appendBehavior',
    'associationBehavior',
    'attachEventBehavior',
    'boundaryEventBehavior',
    'compensateBoundaryEventBehaviour',
    'createBehavior',
    'createDataObjectBehavior',
    'createParticipantBehavior',
    'dataInputAssociationBehavior',
    'dataStoreBehavior',
    'deleteLaneBehavior',
    'detachEventBehavior',
    'dropOnFlowBehavior',
    'eventBasedGatewayBehavior',
    'fixHoverBehavior',
    'groupBehavior',
    'importDockingFix',
    'isHorizontalFix',
    'labelBehavior',
    'layoutConnectionBehavior',
    'messageFlowBehavior',
    'nonInterruptingBehavior',
    'removeElementBehavior',
    'removeEmbeddedLabelBoundsBehavior',
    'removeParticipantBehavior',
    'replaceConnectionBehavior',
    'replaceElementBehaviour',
    'resizeBehavior',
    'resizeLaneBehavior',
    'rootElementReferenceBehavior',
    'spaceToolBehavior',
    'subProcessPlaneBehavior',
    'subProcessStartEventBehavior',
    'textAnnotationBehavior',
    'toggleCollapseConnectionBehaviour',
    'toggleElementCollapseBehaviour',
    'unclaimIdBehavior',
    'updateFlowNodeRefsBehavior',
    'unsetDefaultFlowBehavior'
  ],
  adaptiveLabelPositioningBehavior: [ 'type', AdaptiveLabelPositioningBehavior ],
  appendBehavior: [ 'type', AppendBehavior ],
  associationBehavior: [ 'type', AssociationBehavior ],
  attachEventBehavior: [ 'type', AttachEventBehavior ],
  boundaryEventBehavior: [ 'type', BoundaryEventBehavior ],
  compensateBoundaryEventBehaviour: [ 'type', CompensateBoundaryEventBehavior ],
  createBehavior: [ 'type', CreateBehavior ],
  createDataObjectBehavior: [ 'type', CreateDataObjectBehavior ],
  createParticipantBehavior: [ 'type', CreateParticipantBehavior ],
  dataInputAssociationBehavior: [ 'type', DataInputAssociationBehavior ],
  dataStoreBehavior: [ 'type', DataStoreBehavior ],
  deleteLaneBehavior: [ 'type', DeleteLaneBehavior ],
  detachEventBehavior: [ 'type', DetachEventBehavior ],
  dropOnFlowBehavior: [ 'type', DropOnFlowBehavior ],
  eventBasedGatewayBehavior: [ 'type', EventBasedGatewayBehavior ],
  fixHoverBehavior: [ 'type', FixHoverBehavior ],
  groupBehavior: [ 'type', GroupBehavior ],
  importDockingFix: [ 'type', ImportDockingFix ],
  isHorizontalFix: [ 'type', IsHorizontalFix ],
  labelBehavior: [ 'type', LabelBehavior ],
  layoutConnectionBehavior: [ 'type', LayoutConnectionBehavior ],
  messageFlowBehavior: [ 'type', MessageFlowBehavior ],
  nonInterruptingBehavior: [ 'type', NonInterruptingBehavior ],
  removeElementBehavior: [ 'type', RemoveElementBehavior ],
  removeEmbeddedLabelBoundsBehavior: [ 'type', RemoveEmbeddedLabelBoundsBehavior ],
  removeParticipantBehavior: [ 'type', RemoveParticipantBehavior ],
  replaceConnectionBehavior: [ 'type', ReplaceConnectionBehavior ],
  replaceElementBehaviour: [ 'type', ReplaceElementBehaviour ],
  resizeBehavior: [ 'type', ResizeBehavior ],
  resizeLaneBehavior: [ 'type', ResizeLaneBehavior ],
  rootElementReferenceBehavior: [ 'type', RootElementReferenceBehavior ],
  spaceToolBehavior: [ 'type', SpaceToolBehavior ],
  subProcessPlaneBehavior: [ 'type', SubProcessPlaneBehavior ],
  subProcessStartEventBehavior: [ 'type', SubProcessStartEventBehavior ],
  textAnnotationBehavior: [ 'type', TextAnnotationBehavior ],
  toggleCollapseConnectionBehaviour: [ 'type', ToggleCollapseConnectionBehaviour ],
  toggleElementCollapseBehaviour : [ 'type', ToggleElementCollapseBehaviour ],
  unclaimIdBehavior: [ 'type', UnclaimIdBehavior ],
  unsetDefaultFlowBehavior: [ 'type', UnsetDefaultFlowBehavior ],
  updateFlowNodeRefsBehavior: [ 'type', UpdateFlowNodeRefsBehavior ]
};
